File: backend/src/main/java/com/cg/flightmgmt/service/UserService.java
package com.cg.flightmgmt.service;

import java.util.List;

import com.cg.flightmgmt.dto.User;
import com.cg.flightmgmt.exception.UserNotFoundException;

public interface UserService {
	// create a new user and add it in the database
	public User addUser(User user);

	// view user details from its given user ID
	public User viewUser(int userId) throws UserNotFoundException;

	// view a list of all users
	public List<User> viewUser();

	// update user details
	public User updateUser(User user);

	// delete the user from the database
	public void deleteUser(int userId);

	public int getUserByName(String UserName);
	public boolean validateUser(String userName, String password);
	public String getUserType(int userId);
}

File: Frontend/src/user/service/UserService.js
import axios from 'axios';

class UserService {
    addUser(user){
        return axios.post('http://localhost:5010/airline-api/user/createUser', user);
    }

    getAllUsers(){
        return axios.get('http://localhost:5010/airline-api/user/users');
    }

    getUser(id){
        return axios.get('http://localhost:5010/airline-api/user/users/'+id);
    }

    loginService(userName, password) {
        return axios.post('http://localhost:5010/airline-api/user/login?userName='+userName+'&password='+password);
    }

    getUserType(userId){
        return axios.get('http://localhost:5010/airline-api/user/usertype/'+userId)
    }

    deleteUser(userId){
        return axios.delete('http://localhost:5010/airline-api/user/deleteUser/'+userId);
    }

    getUserByName(name){
        return axios.get('http://localhost:5010/airline-api/user/username/'+name);
    }
}

export default UserService;